lib/shopify_api/plugs/customer_authenticator.ex

```diff
@@ -0,0 +1,161 @@
+defmodule ShopifyAPI.Plugs.CustomerAuthenticator do
+  @moduledoc """
+  The Shopify.Plugs.CustomerAuthenticator plug allows for authentication of a customer call being made from a Shopify shop with a signed payload.
+
+  ## Liquid Template
+
+  You can create the payload for and signature for that this plug will consume with the following `liquid` template:
+  ```liquid
+    {% assign auth_expiry = "now" | date: "%s" | plus: 86400 | date: "%Y-%m-%dT%H:%M:%S.%L%z" %}
```

Comment:
I'm a little concerned about suggesting setting an expiration period of 24h, as this leaves a broad window for an malicious actor to use a stolen token (as the plug has no way to force expiration of individual tokens).

It may be better to encourage a lower default of ~1h, since the tokens are generated per-page-load?